internal/pkg/agent/cmd/enroll.go

```diff
@@ -328,7 +328,7 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command) error {
 		Insecure:             insecure,
 		UserProvidedMetadata: make(map[string]interface{}),
 		Staging:              staging,
-		FixPermissions:       fromInstall,
+		FixPermissions:       false,
```

Comment:
How the agent is initialized differs between Fleet mode and Managed mode. An enrolment is essentially a switch between these two cases:

https://github.com/elastic/elastic-agent/blob/fd26c38120077a91ce81a0caeacf3b6ba7d0abce/internal/pkg/agent/application/application.go#L87-L125

The biggest difference I could spot that could be affected by the permissions of the agent is the creation of the encrypted configuration state store, which is triggered on this path:

https://github.com/elastic/elastic-agent/blob/fd26c38120077a91ce81a0caeacf3b6ba7d0abce/internal/pkg/agent/application/application.go#L101-L102

We would have to look at the permissions it ends up with if we remove this, and then also decide if it matters. The encryption is still there, but we would want to ensure it isn't readable by a non-root user.

Reply:
From my local tests: 
`-rw-------   1 root  wheel     734 Feb 23 15:22 fleet.enc`

Reply:
Great, those are created by the agent running as root so I hoped that would be the case. 

I think the logic to fix the permissions exists to change the ownership to root from the user that downloaded the installation package. This shouldn't apply to the enroll command.

I think this looks good, but let's wait for @blakerouse to take a look before we merge this.

We will also need to test this when the agent is signed, which likely requires merging this and testing a snapshot build. We need to go through the unified release process to get the binary signed. It shouldn't make a difference but we need to confirm.

Reply:
The original change was from Blake here so hopefully he remembers if there was a reason we did this :) https://github.com/elastic/elastic-agent/commit/48a4703cf28ba7fd05d8f2cb1680438514a78736

Reply:
The original change was done to solve this issue - https://github.com/elastic/beats/issues/26213

I got a felling that changing this will result in it not working on Windows.

Reply:
Since we only care about fixing this on Macs, the safest thing to do is to only set this to `false` when `runtime.GOOS == "darwin"` so it only takes effect on Mac.

Reply:
I did the change, if someone can review it, I would appreciate!